scripts/apps/search/components/fields/AssociatedItemsList.tsx

```diff
@@ -14,32 +14,50 @@ interface IProps {
 }
 
 interface IState {
-    associations: Array<IArticle>;
+    types: Array<string>;
+}
+
+interface IRelated {
+    _id: string;
+    type: string;
 }
 
 /**
  * Render icon with count for each item type present in associations.
  */
 export class AssociatedItemsList extends React.Component<IProps, IState> {
-    readonly state = {associations: []};
+    readonly state = {types: []};
 
     componentDidMount() {
         this.getAssociations();
     }
 
     getAssociations() {
         const {content} = this.props.svc;
+        const associations = this.props.item.associations || {};
+        const related = Object.values(associations)
+            .filter((_related) => _related != null);
+        const relatedTypes = related
+            .map((_related: IRelated) => _related.type)
+            .filter((type) => type != null);
+
+        if (relatedTypes.length === related.length) { // types for all
+            this.setState({types: relatedTypes});
+            return;
+        }
 
-        content.fetchAssociations(this.props.item).then((associations) => {
-            this.setState({associations: Object.values(associations)});
+        content.fetchAssociations(this.props.item).then((_associations: Array<IArticle>) => {
```

Comment:
I assume this would fetch not only related items, but media galleries as well. Is that what you want?

Reply:
yes